pkg/node/register.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/kvstore"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// localNodeRegistrationInterval is the interval in which the local
+	// node is re-registered with the kvstore and BPF maps
+	localNodeRegistrationInterval = time.Duration(2) * time.Minute
+
+	// timeout to wait for initial list of nodes
+	listTimeout = time.Duration(30) * time.Second
+)
+
+var (
+	// nodesPath is the path to where nodes are stored in the kvstore
+	nodesPath = path.Join(kvstore.BaseKeyPrefix, "state", "nodes", "v1")
+
+	// controllers contains all controllers required for node management
+	controllers controller.Manager
+)
+
+var registerOnce sync.Once
```

Comment:
Should this be in the var block just above? Also, could you add a comment explaining what registration it guards?